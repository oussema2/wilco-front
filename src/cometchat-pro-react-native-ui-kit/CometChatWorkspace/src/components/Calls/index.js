export { default as CometChatIncomingCall } from './CometChatIncomingCall';
export { default as CometChatOutgoingCall } from './CometChatOutgoingCall';
export { default as CometChatOutgoingDirectCall } from './CometChatOutgoingDirectCall';
export { default as CometChatIncomingDirectCall } from './CometChatIncomingDirectCall';
