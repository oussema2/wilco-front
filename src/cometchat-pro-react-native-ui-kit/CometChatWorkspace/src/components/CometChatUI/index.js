/* eslint-disable import/no-duplicates */
import React, { useEffect, useRef, useState } from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { View, Text } from 'react-native';
import { CometChatGroupListWithMessages } from '../Groups';
import { CometChatUserListWithMessages } from '../Users';
import { CometChatConversationListWithMessages } from '../Chats';
import { CometChatContextProvider } from '../../utils/CometChatContext';
import { CometChatUserProfile } from '../UserProfile';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MCIIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import theme from '../../resources/theme';
import { heightRatio } from '../../utils/consts';


const Tab = createBottomTabNavigator();

function CometChatUI() {
  const [tabs, setTabs] = useState(null);
  const contextRef = useRef(null);

  useEffect(() => {
    checkRestrictions();
  }, []);
  const checkRestrictions = async () => {
    let isChatEnabled = await contextRef.current.state.FeatureRestriction.isRecentChatListEnabled();
    let isGroupListEnabled = await contextRef.current.state.FeatureRestriction.isGroupListEnabled();
    let isUserSettingsEnabled = await contextRef.current.state.FeatureRestriction.isUserSettingsEnabled();
    let isUserListEnabled = await contextRef.current.state.FeatureRestriction.isUserListEnabled();
    let isCallListEnabled = await contextRef.current.state.FeatureRestriction.isCallListEnabled();
    setTabs({
      isChatEnabled,
      isGroupListEnabled,
      isUserSettingsEnabled,
      isUserListEnabled,
      isCallListEnabled,
    });
  };
  return (
    <CometChatContextProvider ref={contextRef}>
      {tabs ? (
        <Tab.Navigator
          screenOptions={({ route }) => ({
            tabBarIcon: ({ color }) => {
              let iconName;

              if (route.name === 'Chats') {
                return (
                  <MCIIcons name="chat" size={25 * heightRatio} color={color} />
                );
              }
              if (route.name === 'More') {
                return (
                  <MaterialIcons
                    name="more-horiz"
                    size={25 * heightRatio}
                    color={color}
                  />
                );
              }
              if (route.name === 'Users') {
                iconName = 'ios-person-circle-sharp';
              } else if (route.name === 'Groups') {
                iconName = 'people';
              }

              // You can return any component that you like here!
              return (
                <Ionicons
                  name={iconName}
                  size={24 * heightRatio}
                  color={color}
                />
              );
            },
          })}
          tabBarOptions={{
            activeTintColor: theme.color.blue,
            inactiveTintColor: 'rgba(0,0,0,0.5)',
            activeBackgroundColor: theme.color.white,
            inactiveBackgroundColor: theme.color.white,
            labelStyle: { fontSize: 12 },
          }}>
          {tabs.isChatEnabled && (
            <Tab.Screen
              name="Chats"
              component={CometChatConversationListWithMessages}
            />
          )}
          {tabs.isUserListEnabled && (
            <Tab.Screen
              name="Users"
              component={CometChatUserListWithMessages}
            />
          )}
          {tabs.isGroupListEnabled && (
            <Tab.Screen
              name="Groups"
              component={CometChatGroupListWithMessages}
            />
          )}
          {tabs.isUserSettingsEnabled && (
            <Tab.Screen name="More" component={CometChatUserProfile} />
          )}
        </Tab.Navigator>
      ) : null}
    </CometChatContextProvider>
  );
}

export default CometChatUI;
