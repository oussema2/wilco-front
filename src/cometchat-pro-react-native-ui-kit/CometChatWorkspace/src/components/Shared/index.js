export { default as CometChatAvatar } from './CometChatAvatar';
export { default as CometChatSharedMedia } from './CometChatSharedMedia';
export { default as CometChatBadgeCount } from './CometChatBadgeCount';
export { default as CometChatBackdrop } from './CometChatBackdrop';
export { default as CometChatUserPresence } from './CometChatUserPresence';
