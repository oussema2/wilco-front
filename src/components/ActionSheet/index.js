export { default as ActionSheet } from './ActionSheet';
