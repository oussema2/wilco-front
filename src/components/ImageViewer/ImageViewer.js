import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { observer } from 'mobx-react-lite';
import { View } from 'react-native';
import ImageView from 'react-native-image-viewing';
import { styles } from './ImageViewer.styles';
import { Pagination } from '../Pagination';

const ImageViewer = ( {
	index, items, visible, onClosePress
} ) => {
	const [ localIndex, setLocalIndex ] = useState( index );

	let renderFooter = ( { imageIndex } ) => (
		items.length > 1 && (
			<View style={styles.dotContainer}>
				<Pagination length={items.length} activeIndex={imageIndex} />
			</View>
		)
	);

	return (
		<ImageView
			images={items}
			imageIndex={index}
			visible={visible}
			onRequestClose={() => onClosePress( localIndex )}
			FooterComponent={renderFooter}
			onImageIndexChange={( _index ) => setLocalIndex( _index )}
			presentationStyle="fullScreen"
			doubleTapToZoomEnabled
		/>
	);
};

ImageViewer.propTypes = {
	index: PropTypes.number,
	items: PropTypes.any,
	visible: PropTypes.bool,
	onClosePress: PropTypes.func
};

ImageViewer.defaultProps = {
	index: 0,
	items: [],
	visible: false,
	onClosePress: () => {}
};

export default observer( ImageViewer );
