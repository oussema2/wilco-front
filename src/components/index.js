export { PrimaryButton } from './PrimaryButton';
export { TextInput } from './TextInput';
