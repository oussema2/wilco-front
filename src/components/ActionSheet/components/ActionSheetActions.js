import React from 'react';
import { observer } from 'mobx-react';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { ActionSheetGradient } from '../../ActionSheetGradient';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetActions = () => {
	const actionSheetService = useActionSheetService();

	return (
		<ActionSheetGradient style={styles.actions}>
			{actionSheetService.actions.map( ( action ) => (
				<ActionSheetButton
					key={action.title}
					{...action}
					onPress={() => {
						actionSheetService.close();
						action.onPress();
					}}
				/>
			) )}
		</ActionSheetGradient>
	);
};

export default observer( ActionSheetActions );
