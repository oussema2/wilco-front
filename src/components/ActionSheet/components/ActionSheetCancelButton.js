import React from 'react';
import { observer } from 'mobx-react';

import { View } from 'react-native';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetCancelButton = () => {
	const actionSheetService = useActionSheetService();

	return (
		<View style={styles.cancelButtonContainer}>
			<ActionSheetButton
				title="Cancel"
				onPress={() => actionSheetService.close()}
			/>
		</View>
	);
};

export default observer( ActionSheetCancelButton );
