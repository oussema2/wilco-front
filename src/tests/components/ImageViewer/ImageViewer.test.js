import * as React from 'react';
import { render } from '@testing-library/react-native';
import { ImageViewer } from '../../../components/ImageViewer';

describe( 'ImageViewer', () => {
	const testID = 'test-ID';
	let sources = [
		{ uri: 'https://example.photo' }
	];

	const setImageViewVisible = jest.fn();

	describe( 'with the default props', () => {
		describe( 'when the Modal is visible', () => {
			it( 'renders the ImageViewer component correctly', () => {
				const component = render( <ImageViewer
					visible
					imageViewerVisible
					setImageViewVisible={setImageViewVisible}
					imageViewerIndex={0}
					items={sources}
				/> );

				expect( component.queryByTestId( 'pagination-testID' ) ).toBeNull();
				expect( component ).toMatchSnapshot();
			} );
		} );

		describe( 'when the Modal is not visible', () => {
			it( 'renders the ImageViewer component correctly', () => {
				const component = render( <ImageViewer
					visible
					imageViewerVisible={false}
					setImageViewVisible={setImageViewVisible}
					items={sources}
					imageViewerIndex={0}
				/> );

				expect( component.queryByTestId( 'pagination-testID' ) ).toBeNull();
				expect( component ).toMatchSnapshot();
			} );
		} );
	} );

	describe( 'when provided more than one photo', () => {
		it( 'renders ImageView component with pagination', () => {
			sources = [
				{ uri: 'https://example.photo' },
				{ uri: 'https://second.example.photo/2' }
			];

			const component = render(
				<ImageViewer
					visible
					testID={testID}
					setImageViewVisible={setImageViewVisible}
					imageViewerVisible
					imageViewerIndex={0}
					items={sources}
				/>
			);

			expect( component.queryByTestId( 'pagination-testID' ) ).not.toBeNull();
			expect( component ).toMatchSnapshot();
		} );
	} );
} );
