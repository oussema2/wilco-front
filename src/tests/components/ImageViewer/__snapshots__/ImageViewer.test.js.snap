// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageViewer when provided more than one photo renders ImageView component with pagination 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={true}
  onRequestClose={[Function]}
  presentationStyle="fullScreen"
  supportedOrientations={
    Array [
      "portrait",
    ]
  }
  transparent={false}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#000",
          "flex": 1,
        },
        Object {
          "backgroundColor": "#000",
          "opacity": 1,
        },
      ]
    }
  >
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
          "width": "100%",
          "zIndex": 1,
        }
      }
    >
      <RCTSafeAreaView
        emulateUnlessSupported={true}
        style={
          Object {
            "alignItems": "flex-end",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 16,
              "left": 16,
              "right": 16,
              "top": 16,
            }
          }
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#00000077",
              "borderRadius": 22,
              "height": 44,
              "justifyContent": "center",
              "marginRight": 8,
              "marginTop": 8,
              "opacity": 1,
              "width": 44,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFF",
                "fontSize": 19,
                "includeFontPadding": false,
                "lineHeight": 22,
                "textAlign": "center",
              }
            }
          >
            âœ•
          </Text>
        </View>
      </RCTSafeAreaView>
    </View>
    <RCTScrollView
      data={
        Array [
          Object {
            "uri": "https://example.photo",
          },
          Object {
            "uri": "https://second.example.photo/2",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      getItemLayout={[Function]}
      horizontal={true}
      initialNumToRender={1}
      initialScrollIndex={0}
      keyExtractor={[Function]}
      maxToRenderPerBatch={1}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      windowSize={2}
    >
      <View>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View>
            <RCTScrollView
              contentContainerStyle={
                Object {
                  "height": 1334,
                }
              }
              maximumZoomScale={1}
              onScroll={[Function]}
              onScrollEndDrag={[Function]}
              pinchGestureEnabled={true}
              scrollEnabled={true}
              scrollEventThrottle={1}
              showsHorizontalScrollIndicator={false}
              showsVerticalScrollIndicator={false}
              style={
                Object {
                  "height": 1334,
                  "width": 750,
                }
              }
            >
              <View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "height": 1334,
                      "justifyContent": "center",
                      "width": 750,
                    }
                  }
                >
                  <ActivityIndicator
                    color="#FFF"
                    size="small"
                  />
                </View>
                <Image
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  nativeID="animatedComponent"
                  onClick={[Function]}
                  onLoad={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  source={
                    Object {
                      "uri": "https://example.photo",
                    }
                  }
                  style={
                    Object {
                      "height": 0,
                      "opacity": 1,
                      "width": 0,
                    }
                  }
                />
              </View>
            </RCTScrollView>
          </View>
        </View>
        <View
          style={
            Object {
              "width": 750,
            }
          }
        />
      </View>
    </RCTScrollView>
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "bottom": 0,
          "position": "absolute",
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
          "width": "100%",
          "zIndex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#00000077",
            "marginBottom": 40,
            "paddingTop": 25,
            "paddingVertical": 30,
            "width": 750,
          }
        }
      >
        <View
          testID="pagination-testID"
        >
          <View
            accessible={false}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "paddingHorizontal": 20,
                  "paddingVertical": 30,
                },
                Object {
                  "flexDirection": "row",
                },
                Object {
                  "backgroundColor": "transparent",
                  "marginTop": 8,
                  "paddingVertical": 0,
                },
              ]
            }
          >
            <View
              accessible={false}
              collapsable={false}
              focusable={false}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginHorizontal": 4,
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#12A3BE",
                    "borderRadius": 5,
                    "height": 8,
                    "opacity": 1,
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                    "width": 8,
                  }
                }
              />
            </View>
            <View
              accessible={false}
              collapsable={false}
              focusable={false}
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginHorizontal": 4,
                  "opacity": 1,
                }
              }
            >
              <View
                collapsable={false}
                nativeID="animatedComponent"
                style={
                  Object {
                    "backgroundColor": "#D3D3D3",
                    "borderRadius": 5,
                    "height": 8,
                    "opacity": 1,
                    "transform": Array [
                      Object {
                        "scale": 1,
                      },
                    ],
                    "width": 8,
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;

exports[`ImageViewer with the default props when the Modal is not visible renders the ImageViewer component correctly 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={true}
  onRequestClose={[Function]}
  presentationStyle="fullScreen"
  supportedOrientations={
    Array [
      "portrait",
    ]
  }
  transparent={false}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#000",
          "flex": 1,
        },
        Object {
          "backgroundColor": "#000",
          "opacity": 1,
        },
      ]
    }
  >
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
          "width": "100%",
          "zIndex": 1,
        }
      }
    >
      <RCTSafeAreaView
        emulateUnlessSupported={true}
        style={
          Object {
            "alignItems": "flex-end",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 16,
              "left": 16,
              "right": 16,
              "top": 16,
            }
          }
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#00000077",
              "borderRadius": 22,
              "height": 44,
              "justifyContent": "center",
              "marginRight": 8,
              "marginTop": 8,
              "opacity": 1,
              "width": 44,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFF",
                "fontSize": 19,
                "includeFontPadding": false,
                "lineHeight": 22,
                "textAlign": "center",
              }
            }
          >
            âœ•
          </Text>
        </View>
      </RCTSafeAreaView>
    </View>
    <RCTScrollView
      data={
        Array [
          Object {
            "uri": "https://example.photo",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      getItemLayout={[Function]}
      horizontal={true}
      initialNumToRender={1}
      initialScrollIndex={0}
      keyExtractor={[Function]}
      maxToRenderPerBatch={1}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      windowSize={2}
    >
      <View>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View>
            <RCTScrollView
              contentContainerStyle={
                Object {
                  "height": 1334,
                }
              }
              maximumZoomScale={1}
              onScroll={[Function]}
              onScrollEndDrag={[Function]}
              pinchGestureEnabled={true}
              scrollEnabled={true}
              scrollEventThrottle={1}
              showsHorizontalScrollIndicator={false}
              showsVerticalScrollIndicator={false}
              style={
                Object {
                  "height": 1334,
                  "width": 750,
                }
              }
            >
              <View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "height": 1334,
                      "justifyContent": "center",
                      "width": 750,
                    }
                  }
                >
                  <ActivityIndicator
                    color="#FFF"
                    size="small"
                  />
                </View>
                <Image
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  nativeID="animatedComponent"
                  onClick={[Function]}
                  onLoad={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  source={
                    Object {
                      "uri": "https://example.photo",
                    }
                  }
                  style={
                    Object {
                      "height": 0,
                      "opacity": 1,
                      "width": 0,
                    }
                  }
                />
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "bottom": 0,
          "position": "absolute",
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
          "width": "100%",
          "zIndex": 1,
        }
      }
    />
  </View>
</Modal>
`;

exports[`ImageViewer with the default props when the Modal is visible renders the ImageViewer component correctly 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={true}
  onRequestClose={[Function]}
  presentationStyle="fullScreen"
  supportedOrientations={
    Array [
      "portrait",
    ]
  }
  transparent={false}
  visible={true}
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#000",
          "flex": 1,
        },
        Object {
          "backgroundColor": "#000",
          "opacity": 1,
        },
      ]
    }
  >
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
          "width": "100%",
          "zIndex": 1,
        }
      }
    >
      <RCTSafeAreaView
        emulateUnlessSupported={true}
        style={
          Object {
            "alignItems": "flex-end",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 16,
              "left": 16,
              "right": 16,
              "top": 16,
            }
          }
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#00000077",
              "borderRadius": 22,
              "height": 44,
              "justifyContent": "center",
              "marginRight": 8,
              "marginTop": 8,
              "opacity": 1,
              "width": 44,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFF",
                "fontSize": 19,
                "includeFontPadding": false,
                "lineHeight": 22,
                "textAlign": "center",
              }
            }
          >
            âœ•
          </Text>
        </View>
      </RCTSafeAreaView>
    </View>
    <RCTScrollView
      data={
        Array [
          Object {
            "uri": "https://example.photo",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      getItemLayout={[Function]}
      horizontal={true}
      initialNumToRender={1}
      initialScrollIndex={0}
      keyExtractor={[Function]}
      maxToRenderPerBatch={1}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      pagingEnabled={true}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      windowSize={2}
    >
      <View>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View>
            <RCTScrollView
              contentContainerStyle={
                Object {
                  "height": 1334,
                }
              }
              maximumZoomScale={1}
              onScroll={[Function]}
              onScrollEndDrag={[Function]}
              pinchGestureEnabled={true}
              scrollEnabled={true}
              scrollEventThrottle={1}
              showsHorizontalScrollIndicator={false}
              showsVerticalScrollIndicator={false}
              style={
                Object {
                  "height": 1334,
                  "width": 750,
                }
              }
            >
              <View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "height": 1334,
                      "justifyContent": "center",
                      "width": 750,
                    }
                  }
                >
                  <ActivityIndicator
                    color="#FFF"
                    size="small"
                  />
                </View>
                <Image
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  nativeID="animatedComponent"
                  onClick={[Function]}
                  onLoad={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  source={
                    Object {
                      "uri": "https://example.photo",
                    }
                  }
                  style={
                    Object {
                      "height": 0,
                      "opacity": 1,
                      "width": 0,
                    }
                  }
                />
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "bottom": 0,
          "position": "absolute",
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
          "width": "100%",
          "zIndex": 1,
        }
      }
    />
  </View>
</Modal>
`;
